### PR TITLE
[WIP] Creating initial structure and content for the 2021 special election

### PR DESCRIPTION
**Pull request by geekygirldawn**
This should not be merged until the following 2 conditions are met:

1. @jberkus to confirm that this meets the Elekto requirements and that it otherwise looks good.
2. Steering to confirm that the instructions and deadlines in the readme meet with your approval.

cc: @bsnchan @pmorie @thisisnotapril @mbehrendt @vaikas

### CODE REVIEW

**Comment by jberkus** [comment]
So I hate to ask this but ...

If the runoff and the vacancy fill are being handled separately (i.e. the same candidates are not eligible for both) then shouldn't we technically have two different voting cards?

**Comment by julz** [comment]
While we're asking annoying questions.... 🙈

Are y'all intending for folks from VMware/Redhat/IBM (other than Scott and Dave, obviously) to run again in this election or is eligibility limited to those who would actually be able to fill the second seat? (It's not clear to me whether a Special Election for a seat which couldn't be filled by the previous election due to vendor limits follows the same candidacy process or a limited one).

(I'm asking mostly for the selfish reason that one election a year is about as much as my little heart can handle 😅, but it also seems like it makes sense to limit the candidates to those that are eligible for the seat, rather than rerunning the same election less than a month later).

**Comment by geekygirldawn** [comment]
> Are y'all intending for folks from VMware/Redhat/IBM (other than Scott and Dave, obviously) to run again in this election or is eligibility limited to those who would actually be able to fill the second seat? (It's not clear to me whether a Special Election for a seat which couldn't be filled by the previous election due to vendor limits follows the same candidacy process or a limited one).

@julz You are welcome to run again, which would mean that if one of the existing Red Hat people stepped down mid-year, and you were the top candidate eligible to fill that seat, then you could end up on the TOC. This is because of the [TOC Election Mechanics](https://github.com/knative/community/blob/main/mechanics/TOC.md), which state that
> Please note that vendor or organization limits do not restrict candidate eligibility. We encourage all interested candidates in running as there may be occurrences where a TOC member steps down and is back-filled by the candidate with the next most votes from the previous election.

But I understand that running in an election like this is stressful and can be time consuming, so if I were you, I would weigh the likelihood of either of the existing Red Hat employees stepping down against how much stress you're willing to endure :)

**Comment by julz** [comment]
Thanks @geekygirldawn! I would love to have been on the TOC for knative but I'd honestly feel pretty silly running twice in two months for a seat I can't win so my current feeling is I will not run again. I'd really like to suggest that we at some point discuss ways to change these rules, at least for future elections, since the current situation seems pretty unfortunate and a bit unfair.

**Comment by geekygirldawn** [comment]
> I'd really like to suggest that we at some point discuss ways to change these rules, at least for future elections, since the current situation seems pretty unfortunate and a bit unfair.

You should bring this up with the Steering Committee, since they own the process. I'm just here to facilitate the election :) I agree with the reasons for having company limits, but the company limits have made this process really painful, especially with so many people working at these few companies, so it's probably worth having some discussions at the SC about how well this is working and whether it can be improved for future elections.

Both of the seats held by people at Red Hat will be up for re-election in the 2022 election, so you'll get a chance to run against them soon!

**Comment by julz** [comment]
> You should bring this up with the Steering Committee, since they own the process.  I'm just here to facilitate the election :) I agree with the reasons for having company limits, but the company limits have made this process really painful, especially with so many people working at these few companies

Yep, sorry, very much didn't mean to suggest this was your fault @geekygirldawn! I created a couple potential ways forward (#621, #620) for steering folks to discuss.

File: elections/2021-TOC2/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+## Purpose
+
+This special election has 2 goals:
+- **Runoff election**
+  - For the 2021 TOC election, we had a tie for one of the 2 TOC seats. This runoff election 
+    is designed to fill the remaining seat due for election this year on the 
+    [Knative Technical Oversight Committee].
+  - This runoff is between 
+    [David Protasowski](candidate-dprotaso.md) and [Scott Nichols](candidate-n3wscott.md).
+  - **Selection Criteria**: Out of these 2 candidates, the one with the most votes in this special election will
```
Comment (jberkus) [inline elections/2021-TOC2/README.md:19]: ... the most preferred candidate in this special election will get that seat.

"most votes" isn't how preference elections work, so we should be clear on that.

File: elections/2021-TOC2/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    seat that needs to be filled with someone who will serve the remainder
+    of his term ending during the 2022 election.
+  - This vacancy could not be filled using candidates from the previous election because after filling the above
+    seat in the runoff election, VMware and Red Hat / IBM will be at the 
+    [maximum number of seats](https://github.com/knative/community/blob/main/mechanics/TOC.md) (2)
+    held by employees from the same organization (or conglomerate, in the case of companies owning
+    each other).
+  - This means that we need to have at least one (preferably more) nominee who does not
+    work at Red Hat, IBM, or VMware.
+  - **Selection Criteria**: The person with the most votes who would not put any single employer over
```
Comment (jberkus) [inline elections/2021-TOC2/README.md:32]: same as above.

File: elections/2021-TOC2/election.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+end_datetime: 2021-07-15 23:59:00
+no_winners: 2
+allow_no_opinion: True
+delete_after: True
+show_candidate_fields:
+  - employer
+election_officers:
+  - geekygirldawn
+  - jberkus
+eligibility: All Knative contributors with more than 50 contributions are eligible to vote
```
Comment (jberkus) [inline elections/2021-TOC2/election.yaml:13]: eligibility: all Knative contributors who were eligible to vote in the original 2021 TOC election are eligible to vote in this one

Comment (jberkus) [reply elections/2021-TOC2/election.yaml:13]: ... since we're not drawing a new list

File: elections/2021-TOC2/README.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  - This vacancy could not be filled using candidates from the previous election because after filling the above
+    seat in the runoff election, VMware and Red Hat / IBM will be at the 
+    [maximum number of seats](https://github.com/knative/community/blob/main/mechanics/TOC.md) (2)
+    held by employees from the same organization (or conglomerate, in the case of companies owning
+    each other).
+  - This means that we need to have at least one (preferably more) nominee who does not
+    work at Red Hat, IBM, or VMware.
+  - **Selection Criteria**: The person with the most votes who would not put any single employer over 
+    the 2 employee maximum will serve the remainder of Grant's term ending during the 2022 election.
+
```
Comment (jberkus) [inline elections/2021-TOC2/README.md:34]: Add:

Nonwithstanding, Knative contributors should rank candidates strictly in order of their preference.  You do not need to pay attention to the above when casting your ballot.

File: elections/2021-TOC2/election_desc.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  election will serve a two (2) year term.
+- **Filling the vacancy** to replace Grant Rodgers who recently stepped down and serve the remainder
+  of his term ending during the 2022 election. This vacancy could not be filled using 
+  candidates from the previous election because after filling the above
+  seat, VMware and Red Hat / IBM will be at the 
+  [maximum number of seats](https://github.com/knative/community/blob/main/mechanics/TOC.md) (2) 
+  held by employees from the same organization (or conglomerate, in the case of companies owning
+  each other). This means that we need to have at least one (preferably more) nominee who does not 
+  work at Red Hat, IBM, or VMware. 
+
```
Comment (jberkus) [inline elections/2021-TOC2/election_desc.md:16]: Add:

Please rank all candidates together according to your preference and evaluation of them as potential TOC members, regardless of company affiliation.  The candidates for the two seats will be selected by the SC, according to the total preference rankings and TOC eligibility rules.

... as in, we don't need them to try to rank people based on company affiliation or anything.  We'll do that.

**Comment by geekygirldawn** [comment]
@jberkus Thanks! I think I incorporated all of your changes in a few additional commits. Have another look and let me know if you see anything else that needs to be updated!

**Comment by jberkus** [comment]
Looks good to go as soon as the SC approves the dates.

**Comment by devguyio** [comment]
Seems to me this PR now is outdated and needs to be updated to reflect #625 changes.

**Comment by geekygirldawn** [comment]
> Seems to me this PR now is outdated and needs to be updated to reflect #625 changes.

Yep! I talked to Brenda about this, but I need to confirm a couple of things with steering before I update the PR.

**Comment by n3wscott** [comment]
I am going to drop out and focus on some other aspects of the project. I can try for ToC another time. I would rather we focus our energies on something else than a runoff election. Thanks!

**Comment by geekygirldawn** [comment]
With the recent decision to remove company limits for TOC members along with Scott's decision above means that we do not need a run-off election or other special election to finish filling the TOC seats for 2021.

**Pull request closed without merging**